src/coreclr/jit/codegencommon.cpp

```diff
@@ -7542,6 +7536,16 @@ void CodeGen::genFnProlog()
 
 #endif // PROFILING_SUPPORTED
 
+    // For OSR we may have a zero-length prolog. That's not supported
+    // when the method must report a generics context,/ so add a nop if so.
+    //
+    if (compiler->opts.IsOSR() && (GetEmitter()->emitGetPrologOffsetEstimate() == 0) &&
+        (compiler->lvaReportParamTypeArg() || compiler->lvaKeepAliveAndReportThis()))
+    {
+        JITDUMP("OSR: prolog was zero length and has generic context to report: adding nop to pad prolog.\n");
+        instGen(INS_nop);
+    }
```

Comment:
For my curiosity, why/where is this not supported?

Reply:
I think it may well be a superfluous assert:
https://github.com/dotnet/runtime/blob/0266f034221188bc7d0156ba76a571c71885c68f/src/coreclr/gcinfo/gcinfoencoder.cpp#L706-L715

The jit only uses `SetPrologSize` if we're reporting a generics context.

Reply:
Ok, perhaps something to relax in another PR.